stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -1615,27 +1623,39 @@ public func withDevice<R>(_ kind: DeviceKind, _ index: UInt = 0,
 /// - Parameters:
 ///   - body: A closure whose TensorFlow operations are to be executed on the
 ///     specified kind of device.
-@inlinable
+@inline(never)
 public func withDefaultDevice<R>(perform body: () throws -> R) rethrows -> R {
   _ThreadLocalState.value.pushDevice(nil)
   let result = try body()
   _ThreadLocalState.value.popDevice()
   return result
 }
 
+
+/// Returns a valid TF device string such as
+/// "/job:localhost/replica:0/task:0/device:CPU:0", which corresponds to the
+/// closest enclosing withDevice() construct.
+/// A return value of nil indicates the absence of withDevice().
 @usableFromInline
-@_cdecl("_swift_tfc_OpSetDeviceFromScope")
-func _TFCOpSetDeviceFromScope(_ op: CTFEOp, _ status: CTFStatus) {
+func getcurrentDeviceName() -> String? {
```

Comment:
- It's a little odd to define a global function for this. It'd be nicer to defined it as a computed property under `_ExecutionContext`.
- Rename to `currentDeviceName` to be more Swifty (avoiding "get").

```swift
extension _ExecutionContext {
  @usableFromInline
  var currentDeviceName: String {
    ...
  }
}
```

Then you can use it as
```swift
_ExecutionContext.global.currentDeviceName
```

Reply:
Done, thanks for the suggestion!

Reply:
lgtm. very nice!